src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  static alive = [];
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= 50;
+    }
+
+    Animal.alive = Animal.alive.filter(
+      (aliveAnimal) => aliveAnimal.health !== 0,
+    );
```

Comment:
[CODE PERFOMANCE] - To improve performance and avoid unnecessary operations, the filter condition in the `Carnivore` class's `bite` method should check for animals with health less than or equal to 0 instead of not equal to 0. Replace `!== 0` with `> 0` to keep only animals that are still alive.